python/pyspark/sql/tests/test_readwriter.py

```diff
@@ -245,6 +246,8 @@ def test_create(self):
             df.writeTo("test_table").using("parquet").create()
             self.assertEqual(100, self.spark.sql("select * from test_table").count())
 
+    @unittest.skipIf(
+        "SPARK_SKIP_CONNECT_COMPAT_TESTS" in os.environ, "Known behavior change in 4.0")
```

Comment:
It seems that `branch-3.5` has different linter rule.

I made a follow-up to recover `branch-3.5` CI.
- https://github.com/apache/spark/pull/46445